ppl/zqd/db/postgresdb/ztests/intake.yaml

```diff
@@ -0,0 +1,85 @@
+tag: services
+
+script: |
+  # Start an authentication enabled zqd, and create 2 separate access tokens
+  # representing different users in different tenants.
+  ZQD_EXTRA_FLAGS="-auth.enabled=true -auth.clientid=testclient -auth.domain=https://testdomain -auth.jwkspath=./auth-public-jwks.json" source startup.sh
+  zapi -h $ZQD_HOST -configpath ./user1 auth store -access \
+    $(gentoken -domain https://testdomain -privatekeyfile ./auth-private-key -keyid testkey -tenantid tenant1 -userid user1)
+  zapi -h $ZQD_HOST -configpath ./user2 auth store -access \
+    $(gentoken -domain https://testdomain -privatekeyfile ./auth-private-key -keyid testkey -tenantid tenant2 -userid user2)
+  zapi -configpath ./user1 -h $ZQD_HOST auth verify
+  zapi -configpath ./user2 -h $ZQD_HOST auth verify
+  zapi_user1="zapi -configpath ./user1 -h $ZQD_HOST"
+  zapi_user2="zapi -configpath ./user2 -h $ZQD_HOST"
+  echo ===
+
+  # Create a space for each user.
+  $zapi_user1 new user1space1
+  USER1_SP1=$($zapi_user1 ls -l -f zng | zq -f text "name=user1space1 | pick id" -)
+  $zapi_user2 new user2space2
+  USER2_SP2=$($zapi_user2 ls -l -f zng | zq -f text "name=user2space2 | pick id" -)
+  echo ===
+
+  # Create an intake for each user, specifying a target space, and verify
+  # the reported ids match, and each user only sees their intakes.
+  USER1_IN1=$($zapi_user1 intake new -f zng -target user1space1 user1intake1 | zq -f text "pick id" -)
+  USER2_IN2=$($zapi_user2 intake new -f zng -target user2space2 user2intake2 | zq -f text "pick id" -)
+  echo user1 intake ls
+  $zapi_user1 intake ls
+  test $($zapi_user1 intake ls -l -f zng | zq -f text "pick target_space_id" -) = $USER1_SP1 && echo user1 space1 intake1 id match
+  echo ===
+  echo user2 intake ls
+  $zapi_user2 intake ls
+  test $($zapi_user2 intake ls -l -f zng | zq -f text "pick target_space_id" -) = $USER2_SP2 && echo user2 space2 intake2 id match
+  echo ===
+
+  # Verify that user1 can't change the name of user2's intake.
+  $zapi_user1 intake update -name "foo" $USER2_IN2 2>/dev/null || echo failed as expected
+  $zapi_user2 intake ls
+  echo ===
```

Comment:
It would be great to have common setup tasks/hooks for a suite of ztests (e.g. beforeEach). I know there's a lot of setup to get these cases but I'd prefer each test case here was its own test. Helps for readability and when a test breaks there's less context for someone unfamiliar with this area of the code to catch up with.

Reply:
Ok, take a look at 453c71d936e10208d71345ce5747d67c09c0b921